### PR TITLE
Configuration for github actions

### PR DESCRIPTION
**Pull request by clbarnes**
See #227

### CODE REVIEW

**Comment by ekohl** [comment]
I briefly pushed the branch to upstream which triggers the tests to actually run.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+on: [push, pull_request]
```
Comment (ekohl) [inline .github/workflows/ci.yaml:1]: When it comes to lists, I'm always a fan of the multi line version
```suggestion
on:
  - push
  - pull_request
```

The same is even more important with `python-version` in the matrix , just so you can easily add or remove entries

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  tox:
```
Comment (ekohl) [inline .github/workflows/ci.yaml:9]: I'd be nice if you could give this step a name.

**Comment by clbarnes** [comment]
Thanks! MacOS fails because the image doesn't have mercurial in it by default. Other than that, seems to be passing on my side!

Before un-drafting this, I'll remove the appveyor and travis configs and add py39.

**Comment by ekohl** [comment]
Just so you're aware of it: I now need to approve and run every time to actually run.

Could you also give it a name at the top level? That way we don't see the the filename but rather something nice. You can add `name: CI` at the very top.

**Comment by clbarnes** [comment]
Don't worry about approving it so it can run here, I'm checking on my own branch and will un-draft when it's passing there. Working on the last failures!

**Comment by clbarnes** [comment]
Green on my end. py39's argparse apparently formats differently to others', so I had to normalise the whitespace when doing string containment checks on those messages.

**Comment by ekohl** [comment]
Perhaps you can squash the commits and leave out the Python 3.9 changes? Then I'd have no problem merging it but I don't have time to look at the 3.9 changes that closely.

**Review by ekohl — APPROVED** [review]
I can still see appveyor run. @c4urself can you check this too?

IMHO we should squash the commits (can do on merge, but we shouldn't forget).

**Comment by clbarnes** [comment]
Updated the readme buttons, and changelog, and squashed. It's weird that appveyor is still doing anything; the repo doesn't have any mentions of it outside of the changelog. Maybe the repo is still registered on the appveyor side?

**Review by ekohl — APPROVED** [review]
That is also my theory but I don't know appveyor.

**Comment by ekohl** [comment]
@florisla this has been open for a long time. Any reason we can't merge this?

**Comment by florisla** [comment]
Much appreciated @clbarnes, this is great work!

**Pull request merged**